client/extensions/wp-job-manager/app/controller.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import analytics from 'lib/analytics';
+import titlecase from 'to-title-case';
+import { getSiteFragment, sectionify } from 'lib/route';
+import { setDocumentHeadTitle as setTitle } from 'state/document-head/actions';
+import { getSelectedSite } from 'state/ui/selectors';
+import { renderWithReduxStore } from 'lib/react-helpers';
+import WPJobManager from './main';
+
+export function settings( context ) {
+	const siteId = getSiteFragment( context.path );
+	const site = getSelectedSite( context.store.getState() );
```

Comment:
Kind of annoying/confusing to have both these definitions here.

Within a React context, we'd always want to get information from Redux state via selectors. That's because we can rely on components to eventually update to get the correct props, even if e.g. `site` information had to be fetched from the network first. In a (non-React) controller like this, we can't assume that -- we only get one shot. Which I think is why `getSiteFragment` is used here -- it just literally extracts the route fragment that looks like a site, without querying any other information from Redux state (or fetching from the network), as does `getSelectedSite`.

Since we only use `siteId` for analytics in here, I think it's fine to keep it. OTOH, `site` is only passed to the `WPJobManager` component, so I'd opt for dropping that, and `connect()` ing that component instead.